src/librustdoc/html/render/print_item.rs

```diff
@@ -1869,107 +1914,122 @@ fn render_union<'a, 'cx: 'a>(
-                fields.iter().filter(|f| matches!(*f.kind, clean::StructFieldItem(..))).count();
-            let has_visible_fields = count_fields > 0;
-            let toggle = should_hide_fields(count_fields);
-            if toggle {
-                toggle_open(&mut w, format_args!("{} fields", count_fields));
-            }
-            for field in fields {
-                if let clean::StructFieldItem(ref ty) = *field.kind {
-                    write!(
-                        w,
-                        "\n{}    {}{}: {},",
-                        tab,
-                        visibility_print_with_space(field.visibility(tcx), field.item_id, cx),
-                        field.name.unwrap(),
-                        ty.print(cx),
-                    );
-                }
-            }
+        if let Some(g) = g {
+            write!(f, "{}", g.print(cx))?;
+        }
 
-            if has_visible_fields {
-                if it.has_stripped_entries().unwrap() {
-                    write!(w, "\n{}    /* private fields */", tab);
+        match ty {
+            None => {
+                let mut buf = Buffer::html();
+                let where_displayed =
+                    g.map(|g| print_where_clause_and_check(&mut buf, g, cx)).unwrap_or(false);
+                write!(f, "{}", buf.into_inner())?;
+
+                // If there wasn't a `where` clause, we add a whitespace.
+                if !where_displayed {
+                    f.write_str(" {")?;
+                } else {
+                    f.write_str("{")?;
                 }
-                write!(w, "\n{}", tab);
-            } else if it.has_stripped_entries().unwrap() {
-                write!(w, " /* private fields */ ");
-            }
-            if toggle {
-                toggle_close(&mut w);
-            }
-            w.write_str("}");
-        }
-        Some(CtorKind::Fn) => {
-            w.write_str("(");
-            for (i, field) in fields.iter().enumerate() {
-                if i > 0 {
-                    w.write_str(", ");
+
+                let count_fields = fields
+                    .iter()
+                    .filter(|item| matches!(*item.kind, clean::StructFieldItem(..)))
+                    .count();
+                let has_visible_fields = count_fields > 0;
+                let toggle = should_hide_fields(count_fields);
+                if toggle {
+                    toggle_open(&mut f, format_args!("{} fields", count_fields));
                 }
-                match *field.kind {
-                    clean::StrippedItem(box clean::StructFieldItem(..)) => write!(w, "_"),
-                    clean::StructFieldItem(ref ty) => {
+                for field in fields {
+                    if let clean::StructFieldItem(ref ty) = *field.kind {
                         write!(
-                            w,
-                            "{}{}",
+                            f,
+                            "\n{}    {}{}: {},",
```

Comment:
Starting from this line, things changed quite a lot. Why were these changes needed?

Reply:
Ok I see. :+1: